app/assets/javascripts/discourse/app/components/footer-nav.gjs

```diff
@@ -0,0 +1,203 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { service } from "@ember/service";
+import { modifier as modifierFn } from "ember-modifier";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import htmlClass from "discourse/helpers/html-class";
+import { postRNWebviewMessage } from "discourse/lib/utilities";
+import { SCROLLED_UP, UNSCROLLED } from "discourse/services/scroll-direction";
+import { bind } from "discourse-common/utils/decorators";
+import not from "truth-helpers/helpers/not";
+
+class FooterNav extends Component {
+  @service appEvents;
+  @service capabilities;
+  @service scrollDirection;
+  @service composer;
+  @service modal;
+
+  @tracked shouldToggleMobileFooter = false;
+  @tracked canGoBack = false;
+  @tracked canGoForward = false;
+
+  currentRouteIndex = 0;
+  routeHistory = [];
+  backForwardClicked = false;
+
+  registerAppEvents = modifierFn(() => {
+    this.appEvents.on("page:changed", this, "_routeChanged");
+
+    return () => {
+      this.appEvents.off("page:changed", this, "_routeChanged");
+    };
+  });
+
+  @bind
+  registerDirectionObserver() {
+    this.scrollDirection.addObserver(
+      "lastScrollDirection",
+      this.toggleMobileFooter
+    );
+  }
+
+  @bind
+  unregisterDirectionObserver() {
+    this.scrollDirection.removeObserver(
+      "lastScrollDirection",
+      this.toggleMobileFooter
+    );
+  }
+
+  @action
+  _routeChanged(route) {
+    // only update route history if not using back/forward nav
+    if (this.backForwardClicked) {
+      this.backForwardClicked = null;
+      return;
+    }
+
+    this.routeHistory.push(route.url);
+    this.currentRouteIndex = this.routeHistory.length;
+    this.setBackForward();
+  }
+
+  get isFooterVisible() {
+    return !this.composer.isOpen && this.shouldToggleMobileFooter;
+  }
+
+  _modalOn() {
+    const backdrop = document.querySelector(".modal-backdrop");
+    if (!backdrop) {
+      return;
+    }
+
+    postRNWebviewMessage(
+      "headerBg",
+      getComputedStyle(backdrop)["background-color"]
```

Comment:
`getComputedStyle` requires styles to be cascaded, and may become a performance bottleneck. Nowadays, the header background is available as a CSS var, so we can do something like

```
document.documentElement.style.getPropertyValue("--header_background")
```

Reply:
I checked, it's hardcoded #111